spring-cloud-gateway-sample/src/test/java/org/springframework/cloud/gateway/sample/GatewaySampleApplicationTests.java

```diff
@@ -153,6 +153,18 @@ public void rewriteResponeBodyObjectWorks() {
 				.consumeWith(result -> assertThat(result.getResponseBody()).isEqualTo("hello"));
 	}
 
+	@Test
+	@SuppressWarnings("unchecked")
+	public void doubleEncoding() {
```

Comment:
I think this test can not certificate SCG double encode URL.UriComponentsBuilder not check URL has encode or not before decode URL

Reply:
I am not sure I understand what you mean. I just added desription for this PR - there you shoud find my reasoning, why I think this test should pass - basically, if client send encoded URL, the upstream should not (under certain circumstances) receive this URL encoded once more - otherwise it reacts to something else, than the original request.